.github/workflows/render-rmarkdown.yaml

```diff
@@ -0,0 +1,35 @@
+on:
+  push:
+    branches:
+      - main
+      - master
+  pull_request:
+    branches:
+      - main
+      - master
+name: Rmarkdown
+
+jobs:
+  render:
+    name: Render Rmarkdown files
+    runs-on: ubuntu-latest
+    env:
+      GITHUB_PAT: ${{ secrets.GITHUB_TOKEN }}
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 2 # This is important to set for `git diff-tree` to work below
```

Comment:
So `fetch-depth: 1` fetches only the most recent revision, `fetch-depth: 0` fetches all history - what does `fetch-depth:2` do? It fetches HEAD^ so that git-diff works?

Reply:
☝️ I think that's right, but honestly don't fully understand the why here.